src/Java.Interop/Java.Interop/JavaPrimitiveArrays.tt

```diff
@@ -108,7 +118,7 @@ namespace Java.Interop {
 		public new unsafe Jni<#= info.TypeModifier #>ArrayElements GetElements ()
 		{
 			var elements = JniEnvironment.Arrays.Get<#= info.JniMarshalType #>ArrayElements (PeerReference, null);
-			return elements == null ? null : new Jni<#= info.TypeModifier #>ArrayElements (PeerReference, elements);
+			return elements == null ? null : new Jni<#= info.TypeModifier #>ArrayElements (PeerReference, elements, JniEnvironment.Arrays.GetArrayLength (PeerReference));
```

Comment:
This doesn't need to use `JniEnvironment.Arrays.GetArrayLength()`. This can just use the `JavaArray<T>.Length` property, aka `Length`:

```csharp
new Jni<#= info.TypeModifier #>ArrayElements (PeerReference, elements, Length)
```

Reply:
`JniArrayElements.Size` needs to be the number of *bytes* for the *entire* array region, so that you can sanely use it without needing to know what the element size is. (*Or*, we add a `JniArrayElements.ElementSize` property, so that it can be computed...).

As such, a "multiply by element size" operation must be performed *somewhere*, either *here*:

```csharp
 new Jni<#= info.TypeModifier #>ArrayElements (PeerReference, elements, Length*sizeof(<#= info.ManagedType #>))
```

*or* within the e.g. `JniBooleanArrayElements(JniObjectReference, Boolean*, int)` constructor:

```csharp
internal unsafe JniBooleanArrayElements (JniObjectReference arrayHandle, Boolean* elements, int elementCount)
    : base ((IntPtr) elements, checked (elementCount*sizeof(Boolean))
```

I don't have a major preference if the total size is computed within the constructor or by the constructor caller, it just needs to be done *somewhere*.